### PR TITLE
Create second Flutter action for builds and integration tests

### PR DESCRIPTION
**Pull request by holtzmak**
Split original Flutter actions into two actions, one for normal pushes and then one for ensuring `master` branch is bug-free as possible. Done in preparation for #82 .

I will look into making a duplicate action for windows environment in the future, but I expect it would likely require installing android emulator and then device onto the windows machine every time the action is run. Luckily, Macs just have iPhones installed 🤷 .

I've tested both actions already (you can see the history in the Actions tab above). No integration tests have been run yet, as none exist yet, so this PR won't be merged until one exists to test out.

### CODE REVIEW

File: .github/workflows/flutter-action-build-and-integration-testing.yml
Comment (holtzmak) [inline .github/workflows/flutter-action-build-and-integration-testing.yml:23]: This will have to update over time

File: .github/workflows/flutter-action-build-and-integration-testing.yml
Comment (holtzmak) [inline .github/workflows/flutter-action-build-and-integration-testing.yml:38]: Grabbing devices as per [this example](https://medium.com/flutter-community/run-flutter-driver-tests-on-github-actions-13c639c7e4ab)

File: .github/workflows/flutter-action-build-and-integration-testing.yml
Comment (holtzmak) [inline .github/workflows/flutter-action-build-and-integration-testing.yml:57]: I would guess we want integration tests before build, but because these tests use an active GUI, the build had better pass before we launch the application for testing.

Comment (holtzmak) [reply .github/workflows/flutter-action-build-and-integration-testing.yml:57]: At this time, we'll have to add each integration test here when it's ready to be run with the group. 

I've been unable to find a group run command for integration tests. I would have thought that dropping the `--target` or just having `--target=tests_folder` would be sufficient, but it's not.

I expect there is a way, but the Flutter docs have recently updated to this new `integration_test` method of driving integration tests as described [here](https://pub.dev/packages/integration_test#usage).

File: .github/workflows/flutter-action-formatting-and-unit-testing.yml
Comment (holtzmak) [inline .github/workflows/flutter-action-formatting-and-unit-testing.yml:14]: * Rubs hands greedily * Moved this here so each step can have a label, which will help when trying to figure why the build failed.

**Review by MummenRider — APPROVED** [review]
Changes made here LGTM

**Comment by holtzmak** [comment]
Will merge once #104 is merged

**Pull request merged**
